src/gluonnlp/utils/registry.py

```diff
@@ -0,0 +1,174 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Create a registry."""
+
+from typing import Optional, List
+import json
+from json import JSONDecodeError
+
+
+class Registry:
```

Comment:
how do we merge this with existing registries in mxnet?

Reply:
This is a reusable registry class and there is no plan to merge this into MXNet.

Reply:
why do we require yet another registry? what's missing in the existing ones?

Reply:
Previously, we need to create the `registry.py` file and reuse MXNet registry: https://github.com/dmlc/gluon-nlp/blob/5dc6b9c9fab9e99b155554a50466c514b879ea84/src/gluonnlp/data/registry.py#L32. Now, we can just call `MODEL_REGISTRY = Registry('model')`, `TOKENIZER_REGISTRY = Registry('tokenizer')` to support multiple use-cases.